infer/environment.cc

```diff
@@ -1327,7 +1327,19 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
                         }
                         break;
                     case cfg::CFG::MIN_LOOP_LET:
```

Comment:
I rarely get the chance to look at how loop pinning works. Would you mind summarizing for me what it means when we're in the `CFG::MIN_LOOP_LET` case?

Reply:
As far as I understand it, I think that the `minLoops` field on each variable ordinarily tracks the minimum loop nesting depth at which that variable is ever referenced (I think blocks also count as "loops" here, but I'm not sure). This is tracked so we can enforce (in the `default` case of this switch) the restriction on changing a variable's type inside a loop, and is set for each variable at CFG build time.

If a variable is `let`-pinned, however, the restriction is different: we _are_ allowed to change the type of the variable, as long as the new type is a subtype of the pinned type. In that case we don't need to track `minLoops` at all, so we can save a little memory by encoding the fact that it is `let`-pinned as a special (negative) constant stored in `minLoops`.